vscode/src/client/tracer.ts

```diff
@@ -0,0 +1,65 @@
+// Modified from https://github.com/microsoft/vscode-languageserver-node/blob/main/client/src/common/client.ts
+
+import * as vscode from "vscode";
+import { Logger, ResponseError, Tracer } from "vscode-jsonrpc/node";
+
+function isString(value: any): value is string {
+  return typeof value === "string" || value instanceof String;
+}
+
+function data2String(data: object): string {
+  if (data instanceof ResponseError) {
+    const responseError = data as ResponseError<any>;
+    return `  Message: ${responseError.message}\n  Code: ${responseError.code} ${responseError.data ? "\n" + responseError.data.toString() : ""}`;
+  }
+  if (data instanceof Error) {
+    if (isString(data.stack)) {
+      return data.stack;
+    }
+    return (data as Error).message;
+  }
+  if (isString(data)) {
+    return data;
+  }
+  return data.toString();
+}
+
+const tracerToLogOutputChannel = (channel: vscode.LogOutputChannel): Tracer => {
+  const logTrace = (message: string, data?: any) => {
+    channel.info(message);
+    if (data) {
+      channel.info(data2String(data));
```

Comment:
@sjd78 
why we are not using `logObjectTrace` here?

Reply:
That's just what was in the code I borrowed it from, so I didn't think very hard about it.  :man_shrugging: